packages/ai-jsx/src/lib/openai.tsx

```diff
@@ -353,14 +411,28 @@ export async function* OpenAIChatModel(
           return {
             role: 'user',
             content: await render(message.element),
-            name: message.element.props.name,
           };
         case 'assistant':
           return {
             role: 'assistant',
             content: await render(message.element),
           };
         case 'functionCall':
+          if (message.element.props.id && !unmatchedIds.has(message.element.props.id)) {
+            // N.B. Adjacent tool calls will be merged below.
+            return {
+              role: 'assistant',
+              content: '',
+              tool_calls: [
+                {
+                  type: 'function',
+                  function: { name: message.element.props.name, arguments: JSON.stringify(message.element.props.args) },
+                  id: message.element.props.id,
+                },
+              ],
+            };
+          }
+
```

Comment:
GH doesn't allow me to link it, but on on line 367:439, we still emit an assistant message with a `function_call`. Can't we merge all of them to only have `tool_calls`? Or do we have to support both for legacy reasons?